executor/src/test/java/org/k8loud/executor/service/ValidationServiceTest.java

```diff
@@ -0,0 +1,69 @@
+package org.k8loud.executor.service;
+
+import data.ExecutionRQ;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.k8loud.executor.exception.ValidationException;
+import org.k8loud.executor.exception.code.ValidationExceptionCode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static data.ExecutionRQ.createExecutionRQ;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@SpringBootTest
+public class ValidationServiceTest {
+    private static final Map<String, String> VALID_PARAMS = Map.of("param1", "val1");
+    @Autowired
+    private ValidationService validationService;
+
+    @ParameterizedTest
+    @MethodSource("provideValidExecutionRQs")
+    void testValidMapping(String collectionName, String actionName, Map<String, String> params) throws ValidationException {
+        // given
+        ExecutionRQ executionRQ = createExecutionRQ(collectionName, actionName, params);
+
+        // when
+        validationService.validate(executionRQ);
+
+        // then
+        // No exception should be thrown
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideInvalidExecutionRQs")
+    void testInvalidMapping(String collectionName, String actionName, Map<String, String> params, ValidationExceptionCode exceptionCode) {
+        // given
+        ExecutionRQ executionRQ = createExecutionRQ(collectionName, actionName, params);
+
+        try {
```

Comment:
what about using assertj3 to check exception? You can use `exception = catchThrowable(()-><func>)` and then `AssertThat(exception).isExaclyInstanceOf(<exception>.class)`. Probably we'll be able to check with that the `exitCode`

Reply:
The method suggested by you is a better approach, more readable.